apps/sms/views/conversation/js/conversation.js

```diff
@@ -1763,22 +1763,22 @@ var ConversationView = {
       pElement.setAttribute('data-l10n-id', 'no-attachment-text');
     }
 
+    var result;
     if (message.type === 'mms' && !isNotDownloaded && !noAttachment) { // MMS
-      return this.mmsContentParser(message).then((mmsContent) => {
-        pElement.appendChild(mmsContent);
-        return messageDOM;
-      });
+      result = SMIL.parse(message).then(
+        (slideArray) => this.createMmsContent(slideArray)
+      ).then(
+        (mmsContent) => pElement.appendChild(mmsContent)
+      );
     }
 
-    return Promise.resolve(messageDOM);
+    return (result || Promise.resolve()).then(() => messageDOM);
```

Comment:
IMO previous two separate ```return``` statements look more readable than ```result || Promise.resolve()).then(() => messageDOM);```, but up to you :)